spec/lib/payment_calculator/ecf/contract/service_fee_calculations_spec.rb

```diff
@@ -7,22 +7,62 @@ class ModuleTestHarness
 end
 
 describe ::PaymentCalculator::Ecf::Contract::ServiceFeeCalculations do
-  it "performs calculations of the service fees" do
-    band_a = double("Band Double",
-                    per_participant: 996.00,
-                    number_of_participants_in_this_band: 2000,
-                    deduction_for_setup?: true)
-    contract = double("Contract Double",
-                      recruitment_target: 2000,
-                      set_up_fee: 149_651.00,
-                      band_a: band_a,
-                      set_up_recruitment_basis: 2000)
-    call_off_contract = ModuleTestHarness.new({ contract: contract })
+  [
+    {
+      set_up_fee: 149_651.00,
+      band_a_per_participant_price: 996.00,
+      expected_total: 647_149.00,
+      expected_monthly: 22_315.48,
+      expected_per_participant: 323.57,
+    },
+    {
+      set_up_fee: 150_000,
+      band_a_per_participant_price: 995.00,
+      expected_total: 646_000.00,
+      expected_monthly: 22_275.86,
+      expected_per_participant: 323.00,
+    },
+    {
+      set_up_fee: 150_000,
+      band_a_per_participant_price: 1_350.00,
+      expected_total: 930_000.00,
+      expected_monthly: 32_068.97,
+      expected_per_participant: 465.00,
+    },
+    {
+      # service fee variation 1
+      set_up_fee: 0,
+      band_a_per_participant_price: 1_400.00,
+      expected_total: 1_120_000.00,
+      expected_monthly: 38_620.69,
+      expected_per_participant: 560.00,
+    },
+    {
+      # service fee variation 2
+      set_up_fee: 150_000,
+      band_a_per_participant_price: 1_400.00, # unchanged from 1
+      expected_total: 970_000.00,
+      expected_monthly: 33_448.28,
+      expected_per_participant: 485.00, # reduced from 1 by service fee component
+    },
```

Comment:
I am not sure I like those code comments.

Do you think it would make sense to make each of those examples a `describe` or a `context`, which lets you specify some more useful descriptions?

Reply:
Yeah I agree it's a little smelly. As you can tell I was struggling a bit to express intent while also keeping clean structures

Reply:
Yeah. 
I would much prefer something like 
```
context "when conditions for variation 2 are met" # maybe say what they are explicitly, I don't actually know
  let(:set_up_fee) { 150_000 }
  let(:band_a_per_participant_price) { 1_400.00 }

  it "correctly reduces expected value per participant" do
    run_test_with_expected_values(expected_total: 970_000.00, expected_monthly: 33_448.28, expected_per_participant: 485.00)
  end
end
```

Something like that?

Reply:
I'll try it as describes and see how it feels....

Reply:
How about this? https://github.com/DFE-Digital/early-careers-framework/pull/723/commits/b253495aed3fc3dfd7d37344c5d3bf915ab993d9

Reply:
Yeah, I think it's clear enough for my taste :)